### DISCUSSION TITLE
When to use Jupyter Notebook

### DISCUSSION THREAD

**Discussion by meganewing** [sr320/course-fish546-2021#21] [Q&A] [answered]
I asked this question last week in class but I still don't fully understand when to use Jupyter Notebook. Should all our codes be created and run through it? If not, what purpose does it serve? Why do we use Jupyter Notebook as opposed to uploading our files to GitHub? I'm just still pretty confused as to why its beneficial to use over other things, and when we should be using it.

**Comment by kubu4** [comment, marked as answer]
Just documenting discussion from class today.

> Should all our codes be created and run through it?

Ideally, yes. 

> what purpose does it serve?

Think of Jupyter notebooks as a recorder for your code. Jupyter notebooks will keep track of everything you've run and what the resulting output(s) were. Also, for the purposes of this class, it provides a means to evaluate and troubleshoot exercises and analysis.

> Why do we use Jupyter Notebook as opposed to uploading our files to GitHub?

These are different things. A Jupyter notebook is actually a file (`.ipynb`). This file can be shared with people and they can open it on their own computer using Jupyter. You will want to upload your Jupyter notebook files to your class GitHub repo. This allows other people to access the contents of your Jupyter notebook. Another bonus when uploading Jupyter notebook files to GitHub is that GitHub can display the formatted Jupyter notebook file in the web browser. (If you want to see a little bit of what the "raw" `.ipynb` file contains, you can run the `head` command on the file in your terminal).
